### PR TITLE
API-22867-Adds encryption to auth headers

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Removes PII in logs
- Bug - passing PII to logs
- Stop passing auth headers to logs
- Dash owns this maintenance

## Related issue(s)
- [API-22867](https://vajira.max.gov/browse/API-22867)

## Testing done

- old behavior: passing PII to logs
- Describe the steps required to verify your changes: rspec

## What areas of the site does it impact?
- logs

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: modules/claims_api/app/workers/claims_api/encrypt_pii.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+module ClaimsApi
+  class EncryptPii
+    extend ActiveSupport::Concern
+
+    def find_claim(claim_id)
+      ClaimsApi::AutoEstablishedClaim.find_by(id: claim_id)
+    end
+
+    def encrypted_headers(claim)
```
Comment (stiehlrod) [inline modules/claims_api/app/workers/claims_api/encrypt_pii.rb:11]: we don't actually need this, but we can do it if we do need it down the road

File: modules/claims_api/app/workers/claims_api/flash_updater.rb [hunk 2]
```diff
@@ -4,22 +4,25 @@
 require 'sentry_logging'
 
 module ClaimsApi
-  class FlashUpdater
+  class FlashUpdater < EncryptPii
     include Sidekiq::Worker
     include SentryLogging
 
-    def perform(user, flashes, auto_claim_id = nil)
+    def perform(auto_claim_id = nil)
```
Comment (stiehlrod) [inline modules/claims_api/app/workers/claims_api/flash_updater.rb:11]: Concern: not sure if perform is called with a claim_id

File: modules/claims_api/app/workers/claims_api/claim_establisher.rb [hunk 3]
```diff
@@ -72,17 +72,17 @@ def queue_special_issues_updater(auth_headers, special_issues_per_disability, au
           code: disability['code'],
           name: disability['name']
         }
-        ClaimsApi::SpecialIssueUpdater.perform_async(bgs_user(auth_headers),
+        ClaimsApi::SpecialIssueUpdater.perform_async(auto_claim.evss_id,
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/claim_establisher.rb:75]: remove `auto_claim.evss_id`. the special issue updater only accepts 3 arguments as well, we provide 4 here.

File: modules/claims_api/app/workers/claims_api/encrypt_pii.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class EncryptPii
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/encrypt_pii.rb:4]: we're no longer encrypting and there is no PII here. I would change this class name to something more meaningful. Maybe something like `ClaimsApi::UpdaterService`?

File: modules/claims_api/app/workers/claims_api/flash_updater.rb [hunk 5]
```diff
@@ -4,11 +4,14 @@
 require 'sentry_logging'
 
 module ClaimsApi
-  class FlashUpdater
+  class FlashUpdater < EncryptPii
     include Sidekiq::Worker
     include SentryLogging
 
-    def perform(user, flashes, auto_claim_id = nil)
+    def perform(flashes, auto_claim_id = nil)
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/flash_updater.rb:11]: can the claim id's be `nil` here? curious if that could ever happen. The first line of the job that kicks these off is: 
`auto_claim = ClaimsApi::AutoEstablishedClaim.find(auto_claim_id)`

File: modules/claims_api/app/workers/claims_api/special_issue_updater.rb [hunk 6]
```diff
@@ -95,7 +98,7 @@ def claim_from_contention_id(claims, contention_id)
     #
     # @param claim [Hash] BGS claim object
     # @param contention_id [Hash(claim_id:, code:, name:)] Identifier to match existing contention
-    # @param special_issues [Array(String)] List of special issues to append
+    # @param claim.special_issues [Array(String)] List of special issues to append
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/special_issue_updater.rb:101]: I believe these are defining the params for the method below. this should remain `special_issues`

File: modules/claims_api/app/workers/claims_api/special_issue_updater.rb [hunk 7]
```diff
@@ -121,7 +124,7 @@ def required_claim_fields(claim, contention_id, special_issues)
...
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/special_issue_updater.rb:127]: same as above.

File: modules/claims_api/app/workers/claims_api/special_issue_updater.rb [hunk 8]
```diff
@@ -19,7 +19,10 @@ class SpecialIssueUpdater
     # @param contention_id [Hash(claim_id:, code:, name:)] Identifier to match existing contention
     # @param special_issues [Array(String)] List of special issues to append
     # @param auto_claim_id [Integer] default: nil
-    def perform(user, contention_id, special_issues, auto_claim_id = nil)
+    def perform(contention_id, special_issues, auto_claim_id)
+      user = bgs_headers(auto_claim_id)
+      return if user.nil?
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/special_issue_updater.rb:24]: we wouldn't want these returns here. a return without an error would tell sidekiq that everything seemed fine and we would end the job with an unsuccessful submission.

File: modules/claims_api/app/workers/claims_api/updater_service.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class UpdaterService
+    extend ActiveSupport::Concern
+
+    def find_claim(claim_id)
```
Comment (acovrig) [inline modules/claims_api/app/workers/claims_api/updater_service.rb:7]: Why not put `claim = ClaimsApi::AutoEstablishedClaim.find(claim_id)` at line 14 instead/

File: modules/claims_api/app/workers/claims_api/special_issue_updater.rb (hunk 8 above)
Comment (acovrig) [reply modules/claims_api/app/workers/claims_api/special_issue_updater.rb:24]: I'd 2nd this with we should decide what to do (raise an exception/log/something) and write a test for it too.

File: modules/claims_api/app/workers/claims_api/flash_updater.rb [hunk 10]
```diff
@@ -4,11 +4,14 @@
...
 module ClaimsApi
-  class FlashUpdater
+  class FlashUpdater < UpdaterService
     include Sidekiq::Worker
     include SentryLogging
 
-    def perform(user, flashes, auto_claim_id = nil)
+    def perform(flashes, auto_claim_id)
+      user = bgs_headers(auto_claim_id)
+      return if user.nil?
```
Comment (FonzMP) [inline modules/claims_api/app/workers/claims_api/flash_updater.rb:13]: this is the same, the return here would end the job successfully.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Review by acovrig — APPROVED** [review]

**Pull request merged**